processing/src/main/java/io/druid/query/groupby/epinephelinae/GroupByQueryEngineV2.java

```diff
@@ -142,24 +143,71 @@ private GroupByQueryEngineV2()
                   public Sequence<Row> apply(final Cursor cursor)
                   {
                     return new BaseSequence<>(
-                        new BaseSequence.IteratorMaker<Row, GroupByEngineIterator>()
+                        new BaseSequence.IteratorMaker<Row, GroupByEngineIterator<?>>()
                         {
                           @Override
                           public GroupByEngineIterator make()
                           {
-                            ColumnSelectorPlus<GroupByColumnSelectorStrategy>[] selectorPlus = DimensionHandlerUtils.createColumnSelectorPluses(
-                                STRATEGY_FACTORY,
-                                query.getDimensions(),
-                                cursor
-                            );
-                            return new GroupByEngineIterator(
+                            ColumnSelectorPlus<GroupByColumnSelectorStrategy>[] selectorPlus = DimensionHandlerUtils
```

Comment:
Extracted the ```if``` clause to check array-based aggregation is applicable as a separate method.